realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -522,42 +521,37 @@ JNIEXPORT void JNICALL Java_io_realm_internal_SharedRealm_nativeRegisterPartialS
     auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     try {
         JStringAccessor class_name(env, j_class_name);               // throws
-        std::string table_name = JStringAccessor(env, j_table_name); // throws
         JStringAccessor query(env, j_query);                         // throws
 
-        JavaGlobalWeakRef j_callback_weak(env, j_callback);
-        JavaGlobalWeakRef j_realm_weak(env, j_realm);
-        JavaGlobalWeakRef j_clazz_weak(env, j_clazz);
+        // The lambda will capture the copied reference and it will be unrefed when the lambda's life cycle is over.
+        // That happens when the Realm is closed or the callback has been triggered once.
+        JavaGlobalRef j_callback_ref(env, j_callback);
```

Comment:
retaining a global reference to the callback will prevent the Realm from being GC'ed, since the callback may outlive the Realm

Reply:
will use `private final WeakReference<SharedRealm> sharedRealm;` in the Callback